rllib/agents/dreamer/dreamer.py

```diff
@@ -130,7 +130,11 @@ def sample(self, batch_size: int):
             index = int(random.randint(0, available))
             episodes_buffer.append(episode[index : index + self.length])
 
-        return SampleBatch.concat_samples(episodes_buffer)
+        batch = {}
+        for k in episodes_buffer[0].keys():
+            batch[k] = np.stack([e[k] for e in episodes_buffer], axis=0)
```

Comment:
can you please also add a lot of comments about why we want to do this manually here. thanks.